### PR TITLE
[SPARK-LLAP-114] Support for complex types, using HIVE-16568

### PR DESCRIPTION
**Pull request by jdere**
## What changes were proposed in this pull request?
Support complex types, using the changes in HIVE-16568 to support complex types in the LLAP input format.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @jdere .
Could you check the build? This should pass the build like [this](https://travis-ci.org/hortonworks-spark/spark-llap/builds/216238065).

File: src/main/scala/com/hortonworks/spark/sql/hive/llap/HS2JDBCWrapper.scala [hunk 1]
```diff
@@ -20,11 +20,15 @@ package com.hortonworks.spark.sql.hive.llap
 import java.net.URI
 import java.sql.{Connection, DatabaseMetaData, Driver, DriverManager, ResultSet, ResultSetMetaData, SQLException}
```
Comment (bikassaha) [inline src/main/scala/com/hortonworks/spark/sql/hive/llap/HS2JDBCWrapper.scala]: @merlintang  We need a corresponding PR for master and 2.1. If this part of the code is not different across branches then this should land in master. Then 2.1 and then 1.6 branches.

Comment (merlintang) [inline src/main/scala/com/hortonworks/spark/sql/hive/llap/HS2JDBCWrapper.scala]: Our current master is 2.1. thus, we need to land it to the 2.2 and 1.6 as well. I can verify the difference.

**Comment by merlintang** [comment]
Where there are some compilation errors: 

[info] Compiling 6 Scala sources to /home/travis/build/hortonworks-spark/spark-llap/target/scala-2.10/classes...
[error] /home/travis/build/hortonworks-spark/spark-llap/src/main/scala/com/hortonworks/spark/sql/hive/llap/RowConverter.scala:32: value getTypeInfo is not a member of org.apache.hadoop.hive.llap.FieldDesc
[error]         val sparkValue = convertValue(llapRow.getValue(idx), colDesc.getTypeInfo)
[error]                                                                      ^
[error] one error found
[error] (compile:compileIncremental) Compilation failed
[error] Total time: 244 s, completed May 9, 2017 12:17:53

**Pull request merged**